### ISSUE TITLE
Have the execution manager populate a suite of variables exposing the inner state of steps 

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-tool_dataflows#164]
This data will be used for a couple things, but typically it could be used in summary steps at the end which say what happened. It will also be used in some advanced steps later https://github.com/catalyst/moodle-tool_dataflows/issues/120 and https://github.com/catalyst/moodle-tool_dataflows/issues/119

### State transition variables
* [x] every time a step changes state record the state and the time that it changed. All timestamps are microtime

Later issues: 
* make a new color for every state https://github.com/catalyst/moodle-tool_dataflows/issues/53

See also https://github.com/catalyst/moodle-tool_dataflows/issues/165

See also
 #226

**Comment by brendanheywood** [comment]
Right now we have output in the run like this:
```
Engine 'sql': Created
Engine 'sql': Initialised. Dry run: No
Engine 'sql', step 'debug': status Blocked
Engine 'sql', step 'sql': status Waiting
Engine 'sql', step 'cron': status Finished
Engine 'sql', step 'sql': status Flowing
Engine 'sql': Finished
Engine 'sql': Finalised
```

This should be one and the same logic, so once this is implement I'd expect to also see all the timestamps and states in the logs as well as the variables.

Also in that example there is a lot of state transitions missing, ie debug is set to blocked but never flows or finishes. Step 'sql' never finishes.

**Comment by keevan** [comment]
The output has been fixed in #185.
